src/components/routes/Entry.tsx

```diff
@@ -1,74 +1,54 @@
-import MetadataResolver, { ResolvedApolloProps } from '../MetadataResolver';
-import QUERY_ENTRY_PROPERTIES from '../../graphql/queries/EntryProperties';
-import QUERY_ENTRY_VALUES from '../../graphql/queries/EntryValues';
 
-type FullProps = ResolvedApolloProps & RouteComponentProps<any>;
+import MetadataResolver, { DataProps } from '../MetadataResolver';
+import { ComponentLoader } from '../../services/ComponentLoader';
+
+import { QUERY_ENTRY_PROPERTIES, QueryMetadata } from '../../graphql/queries/EntryProperties';
+import { QUERY_ENTRY_VALUES, QueryValues, makeDefaultViewConfig } from '../../graphql/queries/EntryValues';
 
 interface State {}
 
-class Entry extends PureComponent<FullProps, State> {
+class Entry extends PureComponent<DataProps<QueryMetadata, QueryValues> & RouteComponentProps<any>, State> {
 
-    static addUnique(value: string, values: Array<string>) {
-        if (value && values.indexOf(value) < 0) {
-            values.push(value);
+    render () {
+        if (this.props.loading) { return <Loading />; }
+        if (!this.props.metadata.dataSetMetadata) {
+            return null;
         }
-    }
-
-    static getValues(components: Array<any>): Array<string> {
-        let values: Array<string> = [];
-        components.forEach(component => Entry.getValue(component, values));
-        return values;
-    }
-
-    static getValue(component: any, values: Array<string>) {
-        switch (component.type) {
-            case COMPONENTS.value:
-            case COMPONENTS.image:
-            case COMPONENTS.link:
-                Entry.addUnique(component.valueKey, values);
-                Entry.addUnique(component.urlKey, values);
-                Entry.addUnique(component.altKey, values);
-                break;
-
-            case COMPONENTS.keyValue:
-                if (component.values) {
-                    component.values.forEach((_component: Component) => Entry.getValue(_component, values));
-                }
-                break;
-            default:
-                break;
+        const collection = this.props.metadata.dataSetMetadata.collection;
+        const collectionList = this.props.metadata.dataSetMetadata.collectionList;
+        if (!collection) {
+            return null;
         }
-    }
 
-    render () {
-        if (this.props.loading) { return <Loading />; }
-        
-        const { collection } = this.props.metadata.dataSetMetadata;
-        if (!collection) { return null; }
+        const idPerUri: {[key: string]: string | undefined} = {};
+        collectionList.items.map(coll => idPerUri[coll.itemType] = coll.collectionId);
+        const componentConfigs = collection.viewConfig.length > 0 ? collection.viewConfig : makeDefaultViewConfig(collection.properties.items, collection.summaryProperties, collectionList.items);
+        if (!this.props.data.dataSets || !this.props.data.dataSets[this.props.match.params.dataSet]) {
```

Comment:
There's a service for this in "services/getDataSetValues.ts" (Function is called "getCollectionValues").
I'd suggest to use and extend that one to incorporate the ```if (!dataSet)``` check also. That way there's only one return null option. (and I'd also suggest a different route instead of returning an empty page -- see previous comment)